File: binarySearch.cpp
#include <iostream>

using namespace std;

int binarySearch(int myarr[], int size, int num){
    int l = 0;
    int r = size-1;
    int mid;

    while(l <= r){
        mid = (l+r)/2;
        if(myarr[mid] == num){
            return mid;
        } else if (myarr[mid] < num){
            l = mid + 1; // +1 para deconsiderar o ponto MID anterior, já que não houve match
        } else if (myarr[mid] > num){
            r = mid - 1; // -1 para deconsiderar o ponto MID anterior, já que não houve match
        }
    }

    return -1;
}

int main(){
    srand(time(0));      //Seed para a geração de valores aleatórios.

    int nums[20];
    int n;
    int result;

    cout << "[ ";
    for(int i = 0; i < 20; i++){
        nums[i] = (rand() % 20) + (i * 20);  //Gerando array com 20 números aleatórios de 0 a 400 de forma crescente e imprimindo-o.
        cout << nums[i] << ' ';
    }
    cout << ']' << endl;

    cin >> n;  // Escolher um número para teste

    result = binarySearch(nums, 20, n);  //Salva o índice de n na variável result e exibe-o se for encontrado.
    result == -1 ? cout << "Nao foi encontrado." : cout << result;

}

File: Algoritmos de Busca/Python/binarySearch.py
import random as rand

def binarySearch(nums, n):
    l, r = 0, len(nums)-1
    while l <= r:
        mid = int((l + r)/2)
        if nums[mid] == n:
            return mid
        elif nums[mid] > n:
            r = mid - 1
        else:
            l = mid + 1
    return -1

nums = [rand.randint(0,20)+(i*20) for i in range(20)]
print(nums)
n = int(input())
result = binarySearch(nums, n)
print(result) if n != -1 else print("Não foi encontrado.")
